universal-application-tool-0.0.1/app/repository/ProgramRepository.java

```diff
@@ -159,40 +164,93 @@ public Program createOrUpdateDraft(Program existingProgram) {
     return database
         .find(Program.class)
         .where()
-        .eq(
-            "name",
-            database.find(Program.class).setId(programId).select("name").findSingleAttribute())
+        .in("name", database.find(Program.class).select("name").where().eq("id", programId).query())
```

Comment:
I'm not a fan of using arrays here for expediency when we never expect to have more than one value.

We can use .findOne() to expect and return a singular value right?

Reply:
`#findOne()` executes the query and returns an actual result

looks like `setMaxRows(1)` does the trick though

Reply:
was unable to get this to work with `eq`, reverted to `in`